include/xsimd/arch/xsimd_avx512f.hpp

```diff
@@ -1380,18 +1380,16 @@ namespace xsimd
 
         // scatter
         template <class A, class T,
-                  typename std::enable_if<std::is_same<uint32_t, T>::value || std::is_same<int32_t, T>::value,
-                                          void>::type>
+                  class = typename std::enable_if<std::is_same<uint32_t, T>::value || std::is_same<int32_t, T>::value, void>::type>
```

Comment:
Missing `class =` make the template always malformed and this function is never called.